docs/ops/pooling/AdaptiveMaxPool_8.md

```diff
@@ -0,0 +1,81 @@
+## AdaptiveMaxPool<a name="AdaptiveMaxPool"></a> {#openvino_docs_ops_pooling_AdaptiveMaxPool_8}
+
+**Versioned name**: *AdaptiveMaxPool-8*
+
+**Category**: *Pooling*
+
+**Short description**: Applies max pooling with adaptive kernel size over the input.
+
+**Detailed description**: This operation calculates the output based on the input and `output_size`.
+The kernel dimensions are calculated using the following formulae for the `NCDHW` input case:
+
+\f[
+\begin{array}{lcl}
+d_{start}&=&floor(i*D_{in}/D_{out})\\
+d_{end}&=&ceil((i+1)*D_{in}/D_{out})\\
+h_{start}&=&floor(j*H_{in}/H_{out})\\
+h_{end}&=&ceil((j+1)*H_{in}/H_{out})\\
+w_{start}&=&floor(k*W_{in}/W_{out})\\
+w_{end}&=&ceil((k+1)*W_{in}/W_{out})
+\end{array}
+\f]
+
+The output is calculated following this formula:
+
+\f[
+Output(i,j,k)=max(Input[d_{start}:d_{end},h_{start}:h_{end},w_{start}:w_{end}])
+\f]
+
+**Attributes**:
+
+* *output_size*
+
+  * **Description**: the target spacial output shape `[L_out]` for 3D input, `[H_out,W_out]` for 4D input,
+    `[D_out,H_out,W_out]` for 5D input.
+  * **Range of values**: positive integer values
+  * **Type**: int[]
+  * **Default value**: None
+  * **Required**: *yes*
+  
+**Inputs**:
+
+*   **1**: 3D, 4D or 5D input tensor of shape `[N,C,L]`, `[N,C,H,W]` or `[N,C,D,H,W]` and type *T*. Required.
+
+**Outputs**:
+  * **1**: Output of type *T* and shape `[N,C,L_out]`, `[N,C,H_out,W_out]` or `[N,C,D_out,H_out,W_out]`.
+  * **2**: Output of type *T_IND* and same shape as 1st output containing indices of elements in 1st output.
```

Comment:
Can the second output be optional? In pytorch, yes. May we need to make it it optional as well and calculate on demand based on separate attribute flag?

Reply:
We don't have optional outputs from operations.